zerver/middleware.py

```diff
@@ -737,3 +737,31 @@ def process_request(self, request: HttpRequest) -> HttpResponse | None:
             return response
 
         return None
+
+
+class BadAPIRequestError(JsonableError):
+    http_status_code: int = 403
+    code: ErrorCode = ErrorCode.BAD_API_REQUEST
+
+    def __init__(self) -> None:
+        pass
+
+    @staticmethod
+    @override
+    def msg_format() -> str:
+        return _("The Zulip API is only accessible over HTTPs.")
+
+
+class BadAPIRequest(MiddlewareMixin):
+    def process_request(self, request: HttpRequest) -> None:
+        # Most HTTP requests are redirected to HTTPS by NGINX, but this is not optimal for
+        # API requests, as it can cause security issues. This middleware is used to handle
+        # unencrypted API requests that are not made over HTTPS. Returning a easy-to-understand
+        # error message that can help clients understand the situation and fix the security
+        # issue.
```

Comment:
"cause security issues" is not specific.

It is always worth re-reading the comment as if you do not know what the change does, and imagining their thought process.  This leaves me knowing it fixes ... some security problem.  But not what that problem is, or why this is the right fix.

```suggestion
        # Most HTTP requests are redirected to HTTPS by NGINX, but API
        # requests should not be, as clients might follow the redirect
        # and accidentally hide the credentials leak inherent in the
        # insecure HTTP request.  Explicitly reject HTTP requests to
        # the API.

Reply:
Right, yeah I should have been more specific here as well. It just felt awkward for me to write the specifics rather than generalize.